core/src/test/java/org/apache/accumulo/core/fate/zookeeper/DistributedReadWriteLockTest.java

```diff
@@ -146,10 +149,69 @@ public void testLock() throws Exception {
     }
   }
 
+  private DistributedLock recover(QueueLock queueLock, FateId fateId, LockRange expectedRange,
+      LockType expectedType) {
+    var dlock = DistributedReadWriteLock.recoverLock(queueLock, fateId);
+    assertEquals(expectedRange, dlock.getRange());
+    assertEquals(expectedType, dlock.getType());
+    return dlock;
+  }
+
+  @Test
+  public void testRanges() {
```

Comment:
You may already have this testing so ignore if I'm missing it, but it would be good to test locking at the boundaries of the ranges. For example, using the existing (E, M]. Could test locking for that range, then try to lock for range (foo, E] which should succeed and try to lock for range (M,bar] which should also succeed.

Reply:
added a test in ffdc5fb1a806c55ed1c2c3c738ecfeaae5116e35